### PR TITLE
Pagination selector and incorrect row count

### PR DESCRIPTION
**Pull request by mkingori**
Here is the bug fix report for issue [1629](https://github.com/decred/dcrdata/issues/1629)
- Removed the pagination option for maximum number of rows per page with less than 10 rows, because it was greyed out and not selectable.
- Ordered the pagination options for maximum number of rows per page numerically.
- Removed the 144 rows per page on pagination options, to prevent overlapping of windows.

___OTHER BUGS ENCOUNTERED___

___Incorrect rows count on /years page___

_Steps to reproduce_:
Go to the table's top right on the [/years](https://explorer.planetdecred.org/years) page.
_Expected result_:
`1 – 5 of 5 rows` due to the 5 rows available.
_Actual result_:
`1 – 5 of 4 rows` despite having 5 rows.
_Cause_:
Data categorized per year which does not align with count of years since the first block was mined in Decred
_Solution_:
Increment row count by 1  on the period between `01-01` and `02-08` of every year.

___Table is broken on the last page of /blocks___

_Steps to reproduce_:
Click the last page of the table on [/blocks](https://explorer.planetdecred.org/blocks) page
_Expected result_:
To show the correct number of rows and highlight the last page as active
_Actual result_:
Shows incorrect number of rows and highlights the second last page as active
_Cause_:
Serverside logic getting faulty when getting records to be displayed in the last page of the table
_Solution_:
Rewrote part of the logic to get the correct number rows to be queried from database.

### CODE REVIEW

**Comment by chappjc** [comment]
Thanks for the thorough analysis and description of the issues and fixes.  We'll review this asap.

**Comment by mkingori** [comment]
Okay. Thank you.

File: views/explorer.tmpl [hunk 1]
```diff
@@ -15,14 +15,16 @@
...
             <div class="d-flex justify-content-between align-items-end">
                 <span class="h2 d-flex pt-4 pb-1 pr-2">
                     Blocks
                     <span class="dcricon-info fs24 ml-3 mt-2"></span>
                 </span>
                 <div class="pb-1 d-flex justify-content-end align-items-end flex-wrap">
                   <span class="fs12 nowrap text-secondary px-2 my-2">
-                      {{intComma (add $Offset 1)}} &ndash; {{intComma (add $Offset .Rows)}} of {{intComma (add .BestBlock 1) }} rows
+                      <!-- {{intComma (add $Offset 1)}} &ndash; {{intComma (add $Offset .Rows)}} of {{intComma (add .BestBlock 1) }} rows -->
+                      {{intComma (add $Offset 1)}} &ndash; {{intComma (add $Offset $rowsCount)}} of {{intComma (add .BestBlock 1) }} rows
```
Comment (buck54321) [inline views/explorer.tmpl]: Can delete the comment. Consider adding a field to the template data instead of using a template variable.

**Comment by mkingori** [comment]
Yes please.

**Comment by mkingori** [comment]
Share the details.

File: views/explorer.tmpl (hunk 1 above)
Comment (mkingori) [reply views/explorer.tmpl]: Okay

Comment (mkingori) [reply views/explorer.tmpl]: I've made the change

**Pull request merged**